### PR TITLE
Sync patch branches

### PR DESCRIPTION
**Pull request by MaryamZi**
## Purpose
Merge the changes from the ballerina-2.0.x to 2201.0.x.

### CODE REVIEW

File: .github/workflows/publish_timestamped_release.yml [hunk 1]
```diff
@@ -5,6 +5,7 @@ on:
     branches:
       - master
       - stage-swan-lake
+      - ballerina-2.0.x
```
Comment (MaryamZi) [inline .github/workflows/publish_timestamped_release.yml:8]:
```suggestion
```

File: .github/workflows/pull_request_ubuntu_build.yml [hunk 2]
```diff
@@ -8,6 +8,7 @@ on:
       - release-stage
       - stage
       - ballerina-1.2.x
+      - ballerina-2.0.x
```
Comment (MaryamZi) [inline .github/workflows/pull_request_ubuntu_build.yml:11]:
```suggestion
```

File: .github/workflows/publish_timestamped_release.yml (hunk 1 above)
Comment (pcnfernando) [inline .github/workflows/publish_timestamped_release.yml]: We'll have to remove the workflow configs of ballerina-2.0.x

Comment (MaryamZi) [reply .github/workflows/publish_timestamped_release.yml]: Removed in b45bcf1.

**Review by gimantha — DISMISSED** [review]

**Review by NipunaRanasinghe — DISMISSED** [review]
Debugger changes LGTM

**Review by lochana-chathura — APPROVED** [review]
NodeBuilder and Typechecker LGTM.

File: compiler/ballerina-lang/src/test/resources/test-repo/bala/hevayo/package_d/0.2.1/java11/package.json [hunk 3]
```diff
@@ -1,3 +1,3 @@
 {
-  "ballerina_version": "2201.0.1"
+  "ballerina_version": "2209.0.1"
```
[the remark this answers is not in the document]
Comment (MaryamZi) [reply compiler/ballerina-lang/src/test/resources/test-repo/bala/hevayo/package_d/0.2.1/java11/package.json]: Removed as per the offline discussion.

**Review by NipunaRanasinghe — APPROVED** [review]
Debugger related changes LGTM

**Review by lochana-chathura — APPROVED** [review]

**Pull request merged**